var/spack/repos/builtin/packages/rose/package.py

```diff
@@ -4,79 +4,288 @@
+            elif "@0.9.10.0:" in spec:
+                # EDG: f4624773cb93dfcc67a00efb5e9e8affeb57fb73
+                f.write("277fea7e1e5305e5b3c86b550a0c1354e8285b96 1523991417")
+            else:
+                raise InstallError("Unknown ROSE version!")
 
     def autoreconf(self, spec, prefix):
-        bash = which('bash')
-        bash('build')
+        #        if not spec.satisfies('@develop'):
+        with working_dir(self.stage.source_path):
+            bash = which("bash")
+            bash("build")
+
+        if "+edg_source" in spec:
+            git = which("git")
+            git(
+                "clone",
+                "rose-dev@rosecompiler1.llnl.gov:rose/edg4x/edg.git",
+                "src/frontend/CxxFrontend/EDG",
+            )
 
     @property
     def languages(self):
         spec = self.spec
         langs = [
-            'binaries' if '+binanalysis' in spec else '',
-            'c' if '+c' in spec else '',
-            'c++' if '+cxx' in spec else '',
-            'java' if '+java' in spec else '',
-            'fortran' if '+fortran' in spec else ''
+            "binaries" if "+binanalysis" in spec else "",
+            "c" if "+c" in spec else "",
+            "c++" if "+cxx" in spec else "",
+            "java" if "+java" in spec else "",
+            "fortran" if "+fortran" in spec else "",
         ]
         return list(filter(None, langs))
 
     def configure_args(self):
         spec = self.spec
-        cc = self.compiler.cc
-        cxx = self.compiler.cxx
-        return [
-            '--disable-boost-version-check',
+
+        if "+mvapich2_backend" in spec:
+            cc = spec["mvapich2"].mpicc
+            cxx = spec["mvapich2"].mpicxx
+        else:
+            cc = spack_cc
+            cxx = spack_cxx
+
+        if spec.satisfies("@0.9.8:"):
+            edg = "4.12"
+        else:
+            edg = "4.9"
+
+        args = [
+            "--disable-boost-version-check",
+            "--enable-edg_version={0}".format(edg),
             "--with-alternate_backend_C_compiler={0}".format(cc),
             "--with-alternate_backend_Cxx_compiler={0}".format(cxx),
-            "--with-boost={0}".format(spec['boost'].prefix),
+            "--with-boost={0}".format(spec["boost"].prefix),
             "--enable-languages={0}".format(",".join(self.languages)),
-            "--with-z3={0}".format(spec['z3'].prefix) if '+z3' in spec else '',
-            '--disable-tests-directory' if '+tests' not in spec else '',
-            '--enable-tutorial-directory={0}'.format('no'),
         ]
 
-    install_targets = ["install-core"]
+        if "+z3" in spec:
+            args.append("--with-z3={0}".format(spec["z3"].prefix))
+        else:
+            args.append("--without-z3")
+
+        if "+tests" in spec:
+            args.append("--enable-tests-directory")
+        else:
+            args.append("--disable-tests-directory")
```

Comment:
Or try something like:
```
args.append("--%sable-tests-directory" % "en" if "+tests" in spec else "dis")
```
This would reduce the number of lines in this section by a factor of 5.  You can even do:
```
args = [
    "--%sable-tests-directory" % "en" if "+tests" in spec else "dis",
    "--%sable-java" % "en" if "+java" in spec else "dis",
    ...
]
```